packages/nouns-contracts/contracts/governance/NounsDAOLogicV2.sol

```diff
@@ -0,0 +1,966 @@
+    }
+
+    /**
+     * @notice Admin function for setting the dynamic quorum linear coefficient
+     * @param newQuorumLinearCoefficient the new linear coefficient, as a fixed point integer with 6 decimals
+     */
+    function _setQuorumLinearCoefficient(uint32 newQuorumLinearCoefficient) external {
+        require(msg.sender == admin, 'NounsDAO::_setQuorumLinearCoefficient: admin only');
+        DynamicQuorumParams memory params = getDynamicQuorumParamsAt(block.number);
+
+        uint32 oldQuorumLinearCoefficient = params.quorumLinearCoefficient;
+        params.quorumLinearCoefficient = newQuorumLinearCoefficient;
+
+        _writeQuorumParamsCheckpoint(params);
+
+        emit QuorumLinearCoefficientSet(oldQuorumLinearCoefficient, newQuorumLinearCoefficient);
+    }
+
+    /**
+     * @notice Admin function for setting the dynamic quorum quadratic coefficient
+     * @param newQuorumQuadraticCoefficient the new quadratic coefficient, as a fixed point integer with 6 decimals
+     */
+    function _setQuorumQuadraticCoefficient(uint32 newQuorumQuadraticCoefficient) external {
+        require(msg.sender == admin, 'NounsDAO::_setQuorumQuadraticCoefficient: admin only');
+        DynamicQuorumParams memory params = getDynamicQuorumParamsAt(block.number);
+
+        uint32 oldQuorumQuadraticCoefficient = params.quorumQuadraticCoefficient;
+        params.quorumQuadraticCoefficient = newQuorumQuadraticCoefficient;
+
+        _writeQuorumParamsCheckpoint(params);
+
+        emit QuorumQuadraticCoefficientSet(oldQuorumQuadraticCoefficient, newQuorumQuadraticCoefficient);
+    }
+
+    /**
+     * @notice Admin function for setting all the dynamic quorum parameters
+     * @param params the new params, see struct definition for more details about each struct member
+     */
+    function _setDynamicQuorumParams(DynamicQuorumParams memory params) public {
+        if (msg.sender != admin) {
+            revert AdminOnly();
+        }
+        if (
+            params.minQuorumVotesBPS < MIN_QUORUM_VOTES_BPS_LOWER_BOUND ||
+            params.minQuorumVotesBPS > MIN_QUORUM_VOTES_BPS_UPPER_BOUND
+        ) {
+            revert InvalidMinQuorumVotesBPS();
+        }
+        if (params.maxQuorumVotesBPS > MAX_QUORUM_VOTES_BPS_UPPER_BOUND) {
+            revert InvalidMaxQuorumVotesBPS();
+        }
+        if (params.minQuorumVotesBPS > params.maxQuorumVotesBPS) {
+            revert MinQuorumBPSGreaterThanMaxQuorumBPS();
+        }
+
+        DynamicQuorumParams memory oldParams;
+        if (quorumParamsCheckpoints.length > 0) {
+            oldParams = getDynamicQuorumParamsAt(block.number);
+        } else {
+            oldParams = DynamicQuorumParams({
+                minQuorumVotesBPS: 0,
+                maxQuorumVotesBPS: 0,
+                quorumVotesBPSOffset: 0,
+                quorumLinearCoefficient: 0,
+                quorumQuadraticCoefficient: 0
+            });
+        }
```

Comment:
Unnecessary `else` block. The default value of each of these is 0.

Reply:
removed in https://github.com/nounsDAO/nouns-monorepo/commit/11dc46ccf886767298934511db674350131ebf99